staging/src/k8s.io/apiserver/pkg/server/options/etcd.go

```diff
@@ -375,6 +375,47 @@ func (s *EtcdOptions) addEtcdHealthEndpoint(c *server.Config) error {
 		return readyCheck()
 	}))
 
+	if len(s.EtcdServersOverrides) != 0 {
+		if err := s.addOverrideEtcdHealthEndpoint(c); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (s *EtcdOptions) addOverrideEtcdHealthEndpoint(c *server.Config) error {
+	var overrideSet = sets.NewString()
+
+	for _, override := range s.EtcdServersOverrides {
+		tokens := strings.Split(override, "#")
+		// multi overrides may point to the same servers
+		// example: ["apps/deployments#s1.example.com;s2.example.com","apps/replicasets#s1.example.com;s2.example.com"]
+		overrideSet.Insert(tokens[1])
```

Comment:
In addition to the ordering issue that @mengqiy is calling out, this isn't covering the case where the same set of servers are provided in different orders. For example, we'll add 2 health checks for the same set of servers if the overrides provided is this:
```
["apps/deployments#s1.example.com;s2.example.com","apps/replicasets#s2.example.com;s1.example.com"]
```
Given we can't create a set with values as a nested set, we could construct a map with a string that is constructed from sorting and then joining the current set of servers. Alongside this map, we could maintain a list that keeps the provided ordering.

Reply:
Use a map now. Updated. Thanks for the suggestion.